BlazarService/src/main/java/com/hubspot/blazar/listener/GitHubStatusVisitor.java

```diff
@@ -71,6 +76,15 @@ public void visit(RepositoryBuild build) throws Exception {
     try {
       repository.createCommitStatus(sha, state, url, description, "Blazar");
     } catch (IOException e) {
+      Optional<GitHubErrorResponse> gitHubErrorResponse = gitHubHelper.extractErrorResponseFromException(e);
```

Comment:
@gchomatas Refactored things and changed the tests per your suggestions.

The block I'm commenting on here fixes a sentry that we've been getting which is the result of GitHub not letting us post more than 100 statuses. Because we can't delete them, I made this just swallow the exception since there is no point in propagating it if we can't post the status.